packages/components/src/core/Tag/index.testUtils.tsx

```diff
@@ -0,0 +1,285 @@
+import { Args } from "@storybook/react";
+import React from "react";
+import Icon from "../Icon";
+import RawTag from "./index";
+import { ExtraTagProps } from "./style";
+
+const SDS_STYLES: ExtraTagProps["sdsStyle"][] = ["rounded", "square"];
+const SDS_TYPES: ExtraTagProps["sdsType"][] = ["primary", "secondary"];
+
+const DISPLAY_CONTENTS: React.CSSProperties = {
+  display: "contents",
+};
+
+const FONT_WEIGHT_NORMAL: React.CSSProperties = {
+  fontWeight: "normal",
+};
+
+const MID_LABEL: React.CSSProperties = {
+  ...FONT_WEIGHT_NORMAL,
+  borderStyle: "solid none none none",
+  gridColumn: "1 / 6",
+  justifySelf: "stretch",
+  paddingTop: 10,
+};
+
+export const AVAILABLE_ICON_OPTIONS = [
+  undefined,
+  <Icon sdsSize="l" sdsIcon="checkCircle" sdsType="button" />,
+  <Icon sdsSize="l" sdsIcon="loading" sdsType="button" />,
+  <WbSunny />,
+  <CheckCircleOutline />,
+];
+
+const iconOptions = AVAILABLE_ICON_OPTIONS.slice(0, 2);
+
+export const HOVER_OPTIONS = [true, false];
+
+export function CommonScreenshotTestDemo({
+  props,
+  colors,
+}: {
+  props: Args;
+  colors: ExtraTagProps["tagColor"][];
+}): JSX.Element {
+  // loop through all SDS_STYLES
+  return (
+    <>
+      {SDS_STYLES.map((sdsStyle) => {
+        return (
+          <PrimaryTagStyle
+            colors={colors}
+            props={props}
+            sdsStyle={sdsStyle}
+            key={sdsStyle}
+          />
+        );
+      })}
+    </>
+  );
+}
+
+// loop through all COLORS + create headers for SDS_STYLES
+function PrimaryTagStyle({
+  colors,
+  sdsStyle,
+  props,
+}: {
+  colors: ExtraTagProps["tagColor"][];
+  sdsStyle: ExtraTagProps["sdsStyle"];
+  props: Args;
+}) {
+  const topLevel: React.CSSProperties = {
+    columnGap: "20px",
+    display: "inline-grid",
+    fontFamily: "sans-serif",
+    marginRight: "50px",
+  };
+
+  const topLabel: React.CSSProperties = {
+    ...FONT_WEIGHT_NORMAL,
+    fontSize: "2em",
+    gridColumn: "1 / 6",
+    marginBottom: 0,
+  };
+
+  return (
+    <div style={topLevel}>
+      <h2 style={topLabel}>
+        Style: <b>{sdsStyle}</b>
+      </h2>
+      {colors.map((color) => {
+        return (
+          <PrimaryTagColor
+            sdsStyle={sdsStyle}
+            color={color}
+            key={String(color)}
+            props={props}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+// loop through all SDS_TYPES + create headers for COLORS
+function PrimaryTagColor({
+  sdsStyle,
+  color,
+  props,
+}: {
+  sdsStyle: ExtraTagProps["sdsStyle"];
+  color: ExtraTagProps["tagColor"];
+  props: Args;
+}) {
+  const h3Label: React.CSSProperties = {
+    ...MID_LABEL,
+    borderWidth: "5px",
+    fontSize: "1.5em",
+    margin: "20px 0 0 0",
+  };
+
+  return (
+    <div style={DISPLAY_CONTENTS}>
+      <h3 style={h3Label}>
+        Color: <b>{color}</b>
+      </h3>
+      {SDS_TYPES.map((sdsType) => {
+        return (
+          <PrimaryTagType
```

Comment:
should we rename `PrimaryTagType ` to just `TagType`, since the loop goes through all `SDS_TYPES`, not just `primary`?

Reply:
Yeah I think so!